libkbfs/tlf_journal.go

```diff
@@ -1379,17 +1447,39 @@ func (j *tlfJournal) getBlockData(id kbfsblock.ID) (
 func (j *tlfJournal) putBlockData(
 	ctx context.Context, id kbfsblock.ID, context kbfsblock.Context, buf []byte,
 	serverHalf kbfscrypto.BlockCryptKeyServerHalf) error {
+	// Since Acquire can block, it should happen outside of the
+	// journal lock.
+	bufLen := int64(len(buf))
+	j.log.CDebugf(ctx, "Acquiring %d bytes for %s", bufLen, j.tlfID)
+	availableBytes, err := j.diskLimiter.Acquire(ctx, bufLen)
```

Comment:
I'm not sure we should be relying on the FS request timeout here, since it could lead to long beach balls for users.  Maybe something much shorter, on the order of a few seconds, would be better?

Also, if the context times out, I'm not sure the error should be `context.DeadlineExceeded`.  Maybe we should return some specific error about having reached our disk space limit (so the UI can show something useful)?

We should probably have a similar (but distinct) nice UI error for when the local disk is full and the put fails, even if we still technically have resources available.  (Separate PR for that is fine though.)